src/main/scala/com/microsoft/hyperspace/index/sources/default/DefaultFileBasedSource.scala

```diff
@@ -259,6 +259,31 @@ class DefaultFileBasedSource(private val spark: SparkSession) extends FileBasedS
         None
     }
   }
+
+  private def filesFromIndex(index: PartitioningAwareFileIndex): Seq[FileStatus] =
+    try {
+      // keep the `asInstanceOf` to force casting or fallback because Databricks `InMemoryFileIndex`
+      // implementation does return a `SerializableFileStatus` instead of the
+      // standard API's `FileStatus`
+      index.allFiles.map(_.asInstanceOf[FileStatus])
+    } catch {
+      case e: ClassCastException if e.getMessage.contains("SerializableFileStatus") =>
+        val dbClassName = "org.apache.spark.sql.execution.datasources.SerializableFileStatus"
+        // scalastyle:off classforname
+        val clazz = Class.forName(dbClassName)
```

Comment:
Can we use `org.apache.spark.util.hyperspace.Utils.classForName` instead?

Reply:
👍